src/components/views/rooms/SearchBar.js

```diff
@@ -33,11 +33,11 @@ module.exports = React.createClass({
     },
 
     onThisRoomClick: function() {
-        this.setState({ scope: 'Room' });
+        this.setState({ scope: 'Room' }, () => this._searchIfQuery());
```

Comment:
Is there a reason there a reason this needs to be in the setState callback? I can't see anywhere where `_searchIfQuery ` depends on the state.

Reply:
Yeah, `_searchIfQuery` calls `onSearch` which uses `scope` from the state.